### ISSUE TITLE
Response from $apply with a groupby does not conform to odata specifications

### ISSUE THREAD

**Issue by jimbromley-avanade** [OData/AspNetCoreOData#802]
**Assemblies affected**
ASP.NET Core OData 8.x

**Describe the bug**
Response from $apply with a groupby does not conform to odata specifications when controller returns ```IActionResult```

```
public IActionResult Get()
```

**Reproduce steps**

Make a request with a apply that has a group by.

**Data Model**
```
public partial class bm20_indicatorvalue
{
    [KeyAttribute()]
    public System.Int64 indicatorvalueid { get; set; }

    public System.String indicatorvaluekey { get; set; }

    public System.Int32 indicatorid { get; set; }

    public System.Double? value { get; set; }

}
```

**EDM (CSDL) Model**
```
<?xml version="1.0" encoding="utf-8"?>
<edmx:Edmx Version="4.0" 
  xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
  <edmx:DataServices>
    <Schema Namespace="Issue420.Model.bm20" 
      xmlns="http://docs.oasis-open.org/odata/ns/edm">
      <EntityType Name="bm20_indicatorvalue">
        <Key>
          <PropertyRef Name="indicatorvalueid" />
        </Key>
        <Property Name="indicatorvalueid" Type="Edm.Int64" Nullable="false" />
        <Property Name="indicatorvaluekey" Type="Edm.String" Nullable="false" />
        <Property Name="indicatorid" Type="Edm.Int32" Nullable="false" />
        <Property Name="value" Type="Edm.Double" />
      </EntityType>
    </Schema>
    <Schema Namespace="Default" 
      xmlns="http://docs.oasis-open.org/odata/ns/edm">
      <EntityContainer Name="Container">
        <EntitySet Name="bm20_indicatorvalue" EntityType="Issue420.Model.bm20.bm20_indicatorvalue" />
      </EntityContainer>
    </Schema>
  </edmx:DataServices>
</edmx:Edmx>
```

**Request/Response**

Request: ```GET ?$apply=groupby((indicatorid),aggregate(value with sum as indicatorvaluesum))```
Response:
```
[
  {
    "indicatorid": 1,
    "indicatorvaluesum": 98
  }
]
```

**Expected behavior**
Request: ```GET ?$apply=groupby((indicatorid),aggregate(value with sum as indicatorvaluesum))```
Response:
```
{
  "@odata.context": "http://localhost:5000/f34f1e34-55ac-458c-b2a3-f94237567488/dwh/$metadata#bm20_indicatorvalue(indicatorid,indicatorvaluesum)",
  "value": [
    {
      "@odata.id": null,
      "indicatorid": 1,
      "indicatorvaluesum": 98.0
    }
  ]
}
```

**Additional context**

This is a breaking change from aspnet to aspnetcore. 

Related odata spec:
http://docs.oasis-open.org/odata/odata-data-aggregation-ext/v4.0/cs02/odata-data-aggregation-ext-v4.0-cs02.html#_Toc376977446

There does seem to be a work around with unknown consequences. Modifying the method in the controller to return an ```IEnumerable<T>``` provides a properly formed response that complies with the odata specification. 

```
    public IEnumerable<bm20_indicatorvalue> Get()
```

**Comment by julealgon** [comment]
Duplicate of:
- https://github.com/OData/AspNetCoreOData/issues/181

TL;DR: Use `ActionResult<T>` and return the raw value instead

**Comment by jimbromley-avanade** [comment]
@julealgon thanks for the reply. This a sufficient duplicate of #181 ... This can close. I'll make an additional comment there that if the Action<T> is necessary for odata to work properly the samples and documentation needs to reflect that.

**Comment by julealgon** [comment]
> If the `Action<T>` is necessary for odata to work properly the samples and documentation needs to reflect that.

Oh I can totally agree with you on this. Hopefully docs will be updated somewhere (documentation for .NetCore OData is a bit all over the place to be honest).

> This a sufficient duplicate of https://github.com/OData/AspNetCoreOData/issues/181 ... However that is closed without taking any action.

This is not correct however. The other issue is still open. Please keep in mind that while I provided _a_ solution there, I'm not part of the OData team and I cannot close issues myself.

Considering that, I'd still recommend closing this one and tracking the updates on #181.
